### PR TITLE
dist/tools/vera++: improve C/C++ support and optimize CI call

### PR DESCRIPTION
**Pull request by jia200x**
### Contribution description
This commit solves some issues and optimizes the check script in order to run Vera++ only once (instead of twice, one for warnings and one for errors).

In concrete, this PR:
- Adds a "max-line-length-warn" in L004 (Line too long) to warn when line length > 80. This paremeter is configurable from `riot_params.txt`. Note that this warning is only a "puts" command, so it won't make Vera++ return error code
- Modifies T003 (some keywoards should be followed by space) in order to differentiate between C and C++ keywords. This should solve  https://github.com/RIOT-OS/RIOT/issues/15626
- Removes the first call to Vera++ (for warnings) and remove unused files.

### Testing procedure
- Check that Vera++ doesn't show false positives (see https://github.com/RIOT-OS/RIOT/issues/15626)
- Check that the warnings for lines > 100 chars are still shown.

### Issues/PRs references
Fixes https://github.com/RIOT-OS/RIOT/issues/15626

_I hope I don't have to touch TCL files again in my whole life_

### CODE REVIEW

File: dist/tools/vera++/scripts/rules/T003.tcl [hunk 1]
```diff
@@ -1,23 +1,32 @@
...
-    return [expr [lsearch $keywords $s] != -1]
+proc isCKeyword {s} {
+    global c_keywords
+    return [expr [lsearch $c_keywords $s] != -1]
+}
+
+proc isCPPKeyword {s} {
+    global cpp_keywords
+    return [expr [lsearch $cpp_keywords $s] != -1]
 }
```
Comment (miri64) [inline dist/tools/vera++/scripts/rules/T003.tcl:30]: Shouldn't this also check for C keywords?

Comment (jia200x) [reply dist/tools/vera++/scripts/rules/T003.tcl:30]: I manually defined the CPP keywords above. The trick is in the _if_ above: if the keyword is a _pure C_ kw or the file is C++ and tje keywords is a C++ kw_, then the token is marked

Comment (miri64) [reply dist/tools/vera++/scripts/rules/T003.tcl:30]: I understood that as much. But C keywords are also C++ keywords, so I would have expected something like

```suggestion
proc isCPPKeyword {s} {
    global cpp_keywords
    return [expr [lsearch $cpp_keywords $s] != -1] || [isCKeyword $s]
}
```

File: dist/tools/vera++/scripts/rules/T003.tcl [hunk 2]
```diff
@@ -38,7 +47,10 @@ foreach f [getSourceFileNames] {
             }
             set state "other"
         } else {
-            if [isKeyword $tokenName] {
+            # Check if it's a C++ file
+            regexp {\.[ch]pp} $f m
+            # C keywords are valid in C++ but not the other way around
+            if {[isCKeyword $tokenName] || ([info exists m] && [isCPPKeyword $tokenName])} {
```
Comment (miri64) [inline dist/tools/vera++/scripts/rules/T003.tcl:53]: But I guess this `||` has the same effect?

Comment (jia200x) [reply dist/tools/vera++/scripts/rules/T003.tcl:53]: yes, that's what I did. It's just the semantics of the isCPPkeyword function :)

File: dist/tools/vera++/scripts/rules/T003.tcl (hunk 1 above)
Comment (jia200x) [reply dist/tools/vera++/scripts/rules/T003.tcl:30]: see below

**Comment by miri64** [comment]
For reference: 

Here is the output of the static tests in current master: https://github.com/RIOT-OS/RIOT/runs/1656061171
And here is the output of the static tests with this PR merged: https://github.com/miri64/RIOT/actions/runs/465973948

**Comment by jia200x** [comment]
> For reference:

Perfect! Thanks!
Since we are invoking Vera++ with `-w`, all errors are shown as warnings. Otherwise, the > 80 warning appears with a "warning" prefix.

**Review by miri64 — APPROVED** [review]
Yepp and there _are_ still valid errors, so ACK!

**Comment by jia200x** [comment]
thanks for the review!

**Pull request merged**